MapboxMobileEvents/MMEEvent.m

```diff
@@ -296,41 +317,41 @@ - (instancetype)init {
-    if (eventAttributes == MMEEvent.nilAttributes) { // special case for initFromCoder
-        self = [super init];
-    }
-    else if ([NSJSONSerialization isValidJSONObject:eventAttributes]) {
-        if (![eventAttributes.allKeys containsObject:MMEEventKeyEvent]) { // is required
-            *error = [NSError errorWithDomain:MMEErrorDomain code:MMEErrorEventInit userInfo:@{
-                MMEErrorDescriptionKey: @"eventAttributes does not contain MMEEventKeyEvent",
-                MMEErrorEventAttributesKey: eventAttributes
-            }];
-            self = nil;
+    @try {
+        if (eventAttributes == MMEEvent.nilAttributes) { // special case for initFromCoder
+            self = [super init];
         }
-        else if (self = [super init]) {
-            @try {
-                _dateStorage = MMEDate.date;
-                NSMutableDictionary* eventAttributesStorage = [eventAttributes mutableCopy];
-
-                if (![eventAttributesStorage.allKeys containsObject:MMEEventKeyCreated]) {
-                    eventAttributesStorage[MMEEventKeyCreated] = [MMEDate.iso8601DateFormatter stringFromDate:_dateStorage];
-                }
-
-                self.attributesStorage = eventAttributesStorage;
-            }
-            @catch(NSException* eventAttributesException) {
-                *error = [NSError errorWithDomain:MMEErrorDomain code:MMEErrorEventInit userInfo:@{
-                    MMEErrorDescriptionKey: @"exception processing eventAttributes",
-                    MMEErrorUnderlyingExceptionKey: eventAttributesException,
+        else if ([NSJSONSerialization isValidJSONObject:eventAttributes]) {
+            if (![eventAttributes.allKeys containsObject:MMEEventKeyEvent]) { // is required
+                *error = [NSError errorWithDomain:MMEErrorDomain code:MMEErrorEventInitMissingKey userInfo:@{
+                    NSLocalizedDescriptionKey: @"eventAttributes does not contain MMEEventKeyEvent",
                     MMEErrorEventAttributesKey: eventAttributes
                 }];
                 self = nil;
             }
+            else if (self = [super init]) {
+                    _dateStorage = MMEDate.date;
+                    NSMutableDictionary* eventAttributesStorage = [eventAttributes mutableCopy];
+
+                    if (![eventAttributesStorage.allKeys containsObject:MMEEventKeyCreated]) {
+                        eventAttributesStorage[MMEEventKeyCreated] = [MMEDate.iso8601DateFormatter stringFromDate:_dateStorage];
+                    }
+
+                    self.attributesStorage = eventAttributesStorage;
+            }
+        }
+        else {
+            *error = [NSError errorWithDomain:MMEErrorDomain code:MMEErrorEventInitInvalid userInfo:@{
+                NSLocalizedDescriptionKey: @"eventAttributes is not a valid JSON Object",
+                MMEErrorEventAttributesKey: eventAttributes
```

Comment:
... this will crash if `eventAttributes` is nil. Perhaps add `?: NSNull.null`

Reply:
The caller would have to get past the `NS_ASSUME_NONNULL_BEGIN` checks, and this particular one is inside a `@try { ... }` block, but one more check can't hurt (also added on line 328)

Reply:
Answer above. Resolving.